ETH-Merge-Bots/Milestones-in-TTD/src/agent.ts

```diff
@@ -0,0 +1,122 @@
+import { BlockEvent, Finding, HandleBlock } from "forta-agent";
+import {
+  getNumberOfDays,
+  getAverageBlockTime,
+  getEstimatedNumberOfBlocksUntilMerge,
+  getAvgBlockDifficulty,
+} from "./utils";
+import { TERMINAL_TOTAL_DIFFICULTY, MILESTONES, ETH_BLOCK_DATA } from "./eth.config";
+import { createFinding, createFinalFinding } from "./findings";
+import BigNumber from "bignumber.js";
```

Comment:
at first I started with many calculations which BigNumber from ethers were rounding many of them. and then I started using bignumber.js. even though it might not be really necessary now, I find it more precise when dealing with calculations.